### PR TITLE
Create Site: Create Site Epilogue

### LINKED ISSUE #8504: Site Creation: Epilogue
Add epilogue screen for Site Creation.

From design:
create_site_epilogue

### PR DESCRIPTION
**Pull request by ScoutHarris**
Fixes #8504 

To test:
- Go through the new site creation process.
- Create a site.
- The 'Congratulations' confetti view will appear for at least 4 seconds. After 4 seconds:
  - If the site is loaded, it will be displayed.
  - If the site is not loaded, the Congratulations view will continue to be displayed until it is.
- Once the site is loaded, it will fade in and the Congratulations view will fade out.
- The 'Configure' button dismisses the view and the new site's details will be shown.
- The 'Write first post' button dismisses the view and the post editor will be shown.

**Designed:**
designed

**Implemented:**
implemented

**Configure option selected:**
configure

**Write Post option selected:**
write_post

### CODE REVIEW

**Comment by ScoutHarris** [comment]
@nheagy - FYI - this is set to merge into https://github.com/wordpress-mobile/WordPress-iOS/pull/8485 so the diff only shows this feature's changes. However, I'll change it to develop when the time comes.

**Comment by folletto** [comment]
Amazing. It looks great! :D 

Two questions:

1. Can we remove the dark masterbar from the preview? There should be a query string to avoid it to display (probably already used in the Themes preview).
2. When "Configure" is clicked is it just a GIF thing or is going to the sites list? If it's not the case, it should go to the My Sites view _for that site_, not the list of sites. Which is what matches the "Configure" idea (the menu of all the things the user can configure).

**Comment by ScoutHarris** [comment]
@folletto 


> Can we remove the dark masterbar from the preview? There should be a query string to avoid it to display (probably already used in the Themes preview).

Ah, I wasn't aware of that. I'll give it a go.

> When "Configure" is clicked is it just a GIF thing or is going to the sites list? If it's not the case, it should go to the My Sites view for that site, not the list of sites. Which is what matches the "Configure" idea (the menu of all the things the user can configure).

OK. That wasn't clear. The design says "My Sites" and your response to my inquiry to have it go to the site details was "No changes from the design blueprint." But it seems we were on the same page all along. :) I'll change it.

**Comment by folletto** [comment]
> The design says "My Sites" and your response to my inquiry to have it go to the site details was "No changes from the design blueprint." But it seems we were on the same page all along.

Haha yes, sorry for the confusion then! 
I meant exactly that, noted for the future :)

**Comment by ScoutHarris** [comment]
@folletto , @iamthomasbishop - 2 changes per request:
- The site's master bar has been removed from the site preview. (Yea, that looks muuuuuch better!)
- 'Configure' now presents the site's details page.

**Comment by folletto** [comment]
You're ace! 🌟

File: WordPress/Classes/ViewRelated/NUX/SiteCreationNavigationController.swift [hunk 1]
```diff
@@ -1,7 +1,30 @@
 import UIKit
 
+@objc protocol SiteCreationNavigationControllerDelegate {
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/SiteCreationNavigationController.swift]: I'm reiterating my big comment above, but I think we should skip this delegate as well. Unless I'm mistaken and there's a technical reason?

File: WordPress/Classes/ViewRelated/NUX/SiteCreationSitePreviewViewController.swift [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    // MARK: - Properties
+
+    var siteUrl: String?
+
+    @IBOutlet weak var siteView: UIView!
+    @IBOutlet weak var congratulationsView: UIView!
+    @IBOutlet weak var congratulationsLabel: UILabel!
+    @IBOutlet weak var siteReadyLabel: UILabel!
+
+    /// Due to build error when adding WKWebView in a storyboard, must create WKWebView
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/SiteCreationSitePreviewViewController.swift]: Ah, interesting.

Maybe we should just create _everything_ programatically? *ducks*

*runs away*

File: WordPress/Classes/ViewRelated/NUX/SiteCreationSitePreviewViewController.swift [hunk 3]
```diff
@@ -0,0 +1,94 @@
+import UIKit
+import WebKit
+
+class SiteCreationSitePreviewViewController: UIViewController {
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/SiteCreationSitePreviewViewController.swift]: Simple and focused class. I like it.

File: WordPress/Classes/ViewRelated/NUX/SiteCreationSitePreviewViewController.swift [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            let siteUrl = siteUrl,
+            let url = URL(string: siteUrl + hideSiteMasterBarUrlParams) {
+                webView.load(URLRequest(url: url))
+        }
+    }
+
+    /// If the site has finished loading and the Congratulations view has shown long enough,
+    /// fade in the site and fade out the Congratulations view.
+    private func showSite() {
+        if siteLoaded == true && timeExpired == true {
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/SiteCreationSitePreviewViewController.swift]: Minor thing, but I think the `== true` are redundant.

File: WordPress/Classes/ViewRelated/NUX/SiteCreationSitePreviewViewController.swift [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+            UIView.animate(withDuration: WPAnimationDurationDefault, animations: {
+                self.siteView.alpha = 1
+                self.congratulationsView.alpha = 0
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/SiteCreationSitePreviewViewController.swift]: ok, this is suble, _but_, halfway through this animation both views are at 50% which means whatever is behind both will be a little visible. If instead `siteView.alpha` is set to 1 before the animation that background will never be visible. What do you think?

**Comment by ScoutHarris** [comment]
OK @nheagy I believe it's ready for another 👀 please kind sir!

File: WordPress/Classes/ViewRelated/NUX/NUXButtonViewController.swift [hunk 6]
```diff
@@ -52,7 +52,8 @@ class NUXButtonViewController: UIViewController {
     // MARK: - Helpers
 
     private func accessibilityIdentifierForString(_ string: String?) -> String {
```
Comment (nheagy) [inline WordPress/Classes/ViewRelated/NUX/NUXButtonViewController.swift]: lol I've got another suggestion I missed the first time.

Moving "string" out of the method name such as `accessibilityIdentifier(for string: String?)` would be a more swifty name.

**Pull request merged**